### ISSUE TITLE
[BUG] Restricted pages are not being indexed in Typo3 10

### ISSUE THREAD

**Issue by robertschramm** [TYPO3-Solr/ext-solr#2634]
**Describe the bug**
The Solr page indexer is not able to index restricted pages (fe_user group set on page). Instead of a 200:found HTTP return and a json formatted result set we get a 403:forbidden and no json return. This is true with the latest TYPO3 version together with the latest Solr extension version.

**To Reproduce**
1. Install a new project with TYPO3 10 via composer mode
2. Choose to have a start page pre configured
3. Install Solr via composer
4. Perform table updates in maintenance module
5. Create a frontend user group via list module
6. Create a new page under your start page which has the access restricted to either your new frontend user group or to "Show at any login"
7. Configure the Solr connection via the site:configuration module, add the Typoscript includes as shown in https://docs.typo3.org/p/apache-solr-for-typo3/solr/11.0/en-us/GettingStarted/ConfigureExtension.html
8. Make sure that the connection to your Solr server is working via the Solr extension menue in TYPO3 backend
9. Now perform your first indexing via Solr extension menue. The restricted page you have created is not being indexed. An error is displayed for your restricted page in the section "Indexing errors"

**Expected behavior**
The restricted page should be indexed successfully.

**Screenshots**
An error (like in screenshot) is thrown by the page indexer.
<img width="1423" alt="New_TYPO3_site__TYPO3_CMS_10_4_3_" src="https://user-images.githubusercontent.com/66314043/83524391-d8cbfc80-a4e3-11ea-8de4-961ec193a87f.png">

The result of the request is a 403:forbidden page in TYPO3 10:
<img width="982" alt="Postman_typo3_10" src="https://user-images.githubusercontent.com/66314043/83526221-64468d00-a4e6-11ea-952e-3e7492a61269.png">

The result of the same reproduction steps in TYPO3 9 is:
<img width="978" alt="Postman_typo3_9" src="https://user-images.githubusercontent.com/66314043/83526246-6b6d9b00-a4e6-11ea-9eb2-d8f7168a90b7.png">

**Used versions:**
 - TYPO3 Version: [10.4.3]
 - EXT:solr Version: [11.0.1]
 - Used Apache Solr Version: [8.5.1]
 - PHP Version: [7.2.1]

**Additional context**
I compared these results versus the results in the latest TYPO3 9 version (also installed via composer mode) with the latest solr extension version. In TYPO3 9 the page indexer request on the restricted page is successful and returns a 200:found status code together with the expected json result.
I checked the internal response in file /typo3conf/ext/solr/Classes/IndexQueue/PageIndexerRequest.php::206 for the page indexing request and found, that a 403:forbidden header is returned from TYPO3 for this page (and no json return). In TYPO3 9 I get a 200:found header and the expected json return with the same Solr extension version 11.0.1.
I assume that TYPO3 doesn´t any longer correctly render the page when the page indexer header from Solr (X-Tx-Solr-Iq - example: {"requestId":"5ed64694995db","item":2,"page":1,"actions":"indexPage","hash":"263ef97696f0c2793fcc13a872dfaca9","loggingEnabled":"0","accessRootline":"4:-2/c:0"}) is set, but rejects the request with a 403:forbidden like a user in frontend would get it. Maybe something changed in the handling of the accessRootline?

**Comment by dkd-kaehm** [comment]
@robertschramm Thanks for detailed description, we will look on this. Any updates are welcome.

**Comment by robertschramm** [comment]
The bug is still present in TYPO3 10.4.4 - does anyone have an idea how to fix this problem?

**Comment by dkd-kaehm** [comment]
I debuuged the scenario. Following Info is currently known:

* The first FE request(for resolving the FE-Groups) is successfull and gets somethink like '-2:0:1'
* The second FE request(indexing request) is not successfull and exits hard with 403 midden in Middleware chain. 
   Seems to be some BC in core, which disturbs the authentication-faking process from EXT:Solr and, leads to hard dieing with 403. 

So there is some trouble with authentication of FE-User/Group.

We have had to grapple with other BC in core to be able to use EXT:solr within TYPO3 10.4.4 at all, so we have no time found this issue. 

We'll fix this in EXT:solr v 11.0.3+

**Comment by georgringer** [comment]
@dkd-kaehm I am on fixing this as well and it requires me to to in middleware something like
```php
    $groups = [1,2,3];
     $this->context->setAspect('frontend.user', GeneralUtility::makeInstance(UserAspect::class, $user, $groups));
```

and in TsFeController::initUserGroups
```php
       $userGroups = array_merge($userGroups, $this->context->getAspect('frontend.user')->getGroupIds());
 ```
